### PR TITLE
Signature changes

### PR DESCRIPTION
**Pull request by marehr**
This PR changes some function signatures and pulls out time computation from the functions onto the callee site.

### CODE REVIEW

File: include/sliding_window/search/write_output_file_parallel.hpp [hunk 1]
```diff
@@ -30,8 +42,8 @@ inline void write_output_file_parallel(worker_t && worker, ibf_t const & ibf, se
         size_t const end = std::min(start + records_per_thread, num_records);
         tasks.emplace_back(std::async(std::launch::async, worker, start, end, records, ibf, arguments, threshold_data));
     }
-    
-    for (auto && task : tasks)
+
+    for (task_future_t & task : tasks)
```
Comment (eaasna) [inline include/sliding_window/search/write_output_file_parallel.hpp:43]: Why `&` instead of `&&` here?

Comment (marehr) [reply include/sliding_window/search/write_output_file_parallel.hpp:43]: Be aware that `&&` has a different meaning for [Forwarding reference](https://en.cppreference.com/w/cpp/language/reference#Forwarding_references) i.e. `auto &&` and explicit types such as `int &&`.

`auto && x = y;` will have one of these two meanings:

1. if `y` is a lvalue (i.e. `int &`, `seqan3::dna4 &`, ...) x is of the same type (i.e. `int &`, `seqan3::dna4 &`, ...)
2. if `y` is something else (i.e. just `int`, or `int &&`) x is of type `decltype(y) &&`, so `int &&` or `int &&`

Whereas `int &&` is a reference to a temporary (rvalue reference) and can extend the lifetime of a temporary. 

The value type of `tasks` is `std::future<...> &`, so it is effectively case 1. and the expression of `for (auto && task : tasks)` was in fact always `for (task_future_t & task : tasks)`.

In this case I would make it explicit that the reference type of `tasks` is `task_future_t &` (a lvalue reference), as we are modifying the state of the underlying future by using get, and we know that `tasks` doesn't return temporaries.

Comment (eaasna) [reply include/sliding_window/search/write_output_file_parallel.hpp:43]: Ah, thank you for the explanation!

**Comment by marehr** [comment]
Can you approve and merge if everything is fine?

**Pull request merged**